### PR TITLE
export Icon component and add documentation

### PR DESCRIPTION
**Pull request by joshfarrant**
## Summary

Export Icon component and add documentation.

## What should reviewers focus on?

- Happy to export the component?
- Happy with the docs in their current state?

## Supporting resources (related issues, external links, etc):

- https://github.com/primer/brand/issues/827#issuecomment-2497358178

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] UI Changes contain new visual snapshots (generated by adding `update snapshots` label to the PR)
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: apps/docs/content/components/Icon.mdx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+<Stack direction="vertical">
+  {namedIconSizes.map((size) => (
+    <Icon
+      key={size}
+      icon={CopilotIcon}
+      size={size}
+      color="green"
+      hasBackground
+    />
+  ))}
```
Comment (rezrah) [inline apps/docs/content/components/Icon.mdx:77]: Can we tweak this example to be more obvious about size values please. `namedIconSizes` is an import in the mdx file and docs-only reader can't see what the allowed values are. We can keep the prop table values the same though.

Screenshot 2024-12-02 at 13 46 59


```suggestion
  <Icon icon={CopilotIcon} size="small" />
  <Icon icon={CopilotIcon} size="medium" />
  <Icon icon={CopilotIcon} size="large" />
```

Comment (joshfarrant) [reply apps/docs/content/components/Icon.mdx:77]: Ah, an oversight on my part there, thanks.

File: .changeset/lovely-jobs-lay.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react-brand': minor
+---
+
+Add new Icon component
```
Comment (rezrah) [inline .changeset/lovely-jobs-lay.md:5]:
```suggestion
Add new `Icon` component
```

**Review by rezrah — APPROVED** [review]
Looks great, ship it 🚀

**Pull request merged**